addons/sale_project/models/sale_order_line.py

```diff
@@ -108,7 +108,7 @@ def create(self, vals_list):
             if line.state == 'sale' and not line.is_expense:
                 line.sudo()._timesheet_service_generation()
                 # if the SO line created a task, post a message on the order
-                if line.task_id:
+                if line.task_id and not self.env.context.get('fsm_mode'):
```

Comment:
After discussing with @IT-Ideas about that, we think it will be better to check if the task_id is not set in the line before calling the `_timesheet_service_generation` method to know if the task is generated or not by this method. 🙂 

And so we should have something like that: 
```suggestion
            if line.state == 'sale' and not line.is_expense:
                has_task = bool(line.task_id)
                line.sudo()._timesheet_service_generation()
                # if the SO line created a task, post a message on the order
                if line.task_id and not has_task:
```

Reply:
@xavierbol , Done!!